src/map/attack.cpp

```diff
@@ -108,33 +108,23 @@ void CAttack::SetCritical(bool value)
             }
         }
 
-        SKILLTYPE skilltype = SKILLTYPE::SKILL_NONE;
+        SKILLTYPE skilltype  = SKILLTYPE::SKILL_NONE;
+        SLOTTYPE  weaponSlot = static_cast<SLOTTYPE>(GetWeaponSlot());
 
         if (m_attacker->objtype == TYPE_PC)
         {
-            SLOTTYPE slot = SLOT_MAIN;
-
-            if (m_attackDirection == PHYSICAL_ATTACK_DIRECTION::RIGHTATTACK)
+            if (auto* weapon = dynamic_cast<CItemWeapon*>(m_attacker->m_Weapons[weaponSlot]))
```

Comment:
weaponSlot is always guaranteed to be sane and within bounds for m_Weapons?

Reply:
> weaponSlot is always guaranteed to be sane and within bounds for m_Weapons?

Yeah It can only return either MAIN or SUB in this context, there is a check above for Daken so it will not return AMMO in this context. Overall it can only return one of those three.